api/src/main/java/jakarta/data/Restrict.java

```diff
@@ -0,0 +1,232 @@
+        return new TextRestrictionRecord<>(field, Operator.LIKE, ESCAPED, pattern);
+    }
+
+    public static <T> Restriction<T> equalTo(Object value, String field) {
+        return new BasicRestrictionRecord<>(field, Operator.EQUAL, value);
+    }
+
+    public static <T> TextRestriction<T> equalTo(String value, String field) {
+        return new TextRestrictionRecord<>(field, Operator.EQUAL, value);
+    }
+
+    public static <T> Restriction<T> greaterThan(Comparable<Object> value, String field) {
+        return new BasicRestrictionRecord<>(field, Operator.GREATER_THAN, value);
+    }
+
+    public static <T> TextRestriction<T> greaterThan(String value, String field) {
+        return new TextRestrictionRecord<>(field, Operator.GREATER_THAN, value);
+    }
+
+    public static <T> Restriction<T> greaterThanEqual(Comparable<Object> value, String field) {
+        return new BasicRestrictionRecord<>(field, Operator.GREATER_THAN_EQUAL, value);
+    }
+
+    public static <T> TextRestriction<T> greaterThanEqual(String value, String field) {
+        return new TextRestrictionRecord<>(field, Operator.GREATER_THAN_EQUAL, value);
+    }
+
+    public static <T> Restriction<T> in(Set<Object> values, String field) {
+        return new BasicRestrictionRecord<>(field, Operator.IN, values);
+    }
+
+    public static <T> Restriction<T> lessThan(Comparable<Object> value, String field) {
+        return new BasicRestrictionRecord<>(field, Operator.LESS_THAN, value);
+    }
+
+    public static <T> TextRestriction<T> lessThan(String value, String field) {
+        return new TextRestrictionRecord<>(field, Operator.LESS_THAN, value);
+    }
+
+    public static <T> Restriction<T> lessThanEqual(Comparable<Object> value, String field) {
+        return new BasicRestrictionRecord<>(field, Operator.LESS_THAN_EQUAL, value);
+    }
+
+    public static <T> TextRestriction<T> lessThanEqual(String value, String field) {
+        return new TextRestrictionRecord<>(field, Operator.LESS_THAN_EQUAL, value);
+    }
+
+    // TODO this would be possible if Pattern is added, but is it even useful?
+    //public static <T> TextRestriction<T> like(Pattern pattern, String field) {
+    //    return new TextRestriction<>(field, Operator.LIKE, ESCAPED, pattern);
+    //}
+
+    public static <T> TextRestriction<T> like(String pattern, String field) {
+        return new TextRestrictionRecord<>(field, Operator.LIKE, pattern);
+    }
+
+    public static <T> TextRestriction<T> like(String pattern,
+                                               char charWildcard,
+                                               char stringWildcard,
+                                               String field) {
+        String p = toLikeEscaped(charWildcard, stringWildcard, false, pattern, false);
+        return new TextRestrictionRecord<>(field, Operator.LIKE, ESCAPED, p);
+    }
+
+    public static <T> Restriction<T> not(Object value, String field) {
```

Comment:
It is not clear that `not` means, `notEquals´. 
 Please either, use `ne` or `NotEquals`

Reply:
Sure, I can update that. Given that we have a `equalTo` method, the equivalent here will be `notEqualTo`